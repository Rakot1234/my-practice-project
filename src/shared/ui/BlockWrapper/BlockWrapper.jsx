import React, { PureComponent } from 'react';
import './BlockWrapper.scss';
import PropTypes from 'prop-types';

class BlockWrapper extends PureComponent {
    static propTypes = {
        className: PropTypes.string,
        bottomBorder: PropTypes.bool,
        children: PropTypes.node,
        innerColor: PropTypes.oneOf(['transparent', 'white', 'gray']),
    };

    static defaultProps = {
        className: '',
        bottomBorder: false,
        innerColor: 'transparent'
    };

    render() {
        const { children, className, bottomBorder, innerColor } = this.props;

        return (
            <div className={
                    `block-wrapper 
                    block-wrapper_bottom-border_${bottomBorder} 
                    block-wrapper_inner-color_${innerColor}`
                }
            >
                <div className={`block-wrapper__inner ${className}`}>
                    {children}
                </div>
            </div>
        );
    }
};

export default BlockWrapper;
