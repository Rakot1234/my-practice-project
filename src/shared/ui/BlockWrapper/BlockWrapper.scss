@import '../../../styles/variables.scss';

$block: block-wrapper;

.#{$block} {
    &__inner {
        max-width: 1170px;
        padding: 50px 0 40px 0;
        margin: auto;
    }
    &_bottom-border_true {
        border-bottom: 1px solid #{$slightGray};
    }
    //inner color mods
    &_inner-color {
        &_transparent {
            background-color: transparent;
        }
        &_white {
            background-color: #{white};   
        }
        &_gray {
            background-color: #{$darkGray};
        }
    }
};
