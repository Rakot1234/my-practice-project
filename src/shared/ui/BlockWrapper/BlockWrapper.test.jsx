import React from 'react';
import { shallow } from 'enzyme';
import BlockWrapper from './BlockWrapper';

describe('<BlockWrapper />', () => {
    it('Should render blockwrapper', () => {
        const wrapper = shallow(<BlockWrapper><div /></BlockWrapper>);
        expect(wrapper).toMatchSnapshot();
    });
});
