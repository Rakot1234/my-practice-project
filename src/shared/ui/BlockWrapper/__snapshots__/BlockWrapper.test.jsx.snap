// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockWrapper /> Should render blockwrapper 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlockWrapper
    bottomBorder={false}
    className=""
    innerColor="transparent"
  >
    <div />
  </BlockWrapper>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="block-wrapper__inner"
      >
        <div />
      </div>,
      "className": "block-wrapper block-wrapper_bottom-border_false block-wrapper_inner-color_transparent",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div />,
        "className": "block-wrapper__inner",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="block-wrapper__inner"
        >
          <div />
        </div>,
        "className": "block-wrapper block-wrapper_bottom-border_false block-wrapper_inner-color_transparent",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "block-wrapper__inner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
